### ISSUE TITLE
Use a custom .jazzy.yml file in order to support grouping in documentation

### ISSUE THREAD

**Issue by orta** [ReactiveX/RxSwift#440]
Example taken from https://github.com/bustoutsolutions/siesta/blob/master/.jazzy.yaml

``` yaml
custom_categories:
  - name: Resources
    children:
    - Service
    - Resource
    - Entity
    - Error
    - Configuration
    - ConfigurationPatternConvertible
    - TypedContentAccessors

  - name: Observering Resources
    children:
    - ResourceObserver
    - ResourceObserverClosure
    - ResourceEvent

  - name: Requests
    children:
    - Request
    - RequestMethod

  - name: Response Parsing
    children:
    - Response
    - ResponseTransformer
    - TransformerSequence
    - ResponseContentTransformer
    - JSONResponseTransformer(_:)
    - TextResponseTransformer(_:)
    - ImageResponseTransformer(_:)

  - name: UI Helpers
    children:
    - ResourceStatusOverlay
    - RemoteImageView

  - name: Network Layer
    children:
    - NetworkingProvider
    - NetworkingProviderConvertible
    - RequestNetworking
    - RequestNetworkingCompletionCallback
    - RequestTransferMetrics
    - NSURLSessionProvider
    - AlamofireProvider

  - name: Caching
    children:
    - EntityCache
    - EntityEncoder
    - JSONEntityEncoder

  - name: Logging
    children:
    - enabledLogCategories
    - LogCategory
    - logger
```

Lets you have custom left-sidebar groupings: http://cocoadocs.org/docsets/Siesta/1.0-beta.6/

![screen shot 2016-01-26 at 20 28 21](https://cloud.githubusercontent.com/assets/49038/12593921/6a4971be-c46b-11e5-9e8d-05f35ef8b386.png)

**Comment by kzaher** [comment]
We'll need to code up another generator from xcode :) Suggestions? How to read that directory structure?

**Comment by kzaher** [comment]
Ok, we have directories, who cares about xcode :) Just need to recurse directories :), my bad. Tired.

**Comment by solidcell** [comment]
I've created a WIP PR for this at #730.

**Comment by solidcell** [comment]
@kzaher this should probably be closed, now that #730 is merged.
